### PR TITLE
feat: Update dependencies and spec support.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java [hunk 1]
```diff
@@ -86,41 +89,120 @@ public Metadata getMetadata() {
     @Override
     public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
         EvaluationDetail<Boolean> detail
-                = this.client.boolVariationDetail(key, evaluationContextConverter.toLdContext(ctx), defaultValue);
+            = this.client.boolVariationDetail(key, evaluationContextConverter.toLdContext(ctx), defaultValue);
```
Comment (kinyoklion) [inline src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java:92]: I feel like formatting always changes even when I am using the same rules and the same tools, but at different points in time.

File: build.gradle [hunk 2]
```diff
@@ -111,11 +122,19 @@ dependencies {
...
-    implementation group: 'com.launchdarkly', name: 'launchdarkly-java-server-sdk', version: '[6.0.0, 7.0.0)'
-    implementation 'dev.openfeature:sdk:[1.2.0,2.0.0)'
+    implementation group: 'com.launchdarkly', name: 'launchdarkly-java-server-sdk', version: '7.1.0-SNAPSHOT'
+//    implementation group: 'com.launchdarkly', name: 'launchdarkly-java-server-sdk', version: '[7.1.0, 8.0.0)'
+    implementation 'dev.openfeature:sdk:[1.7.0,2.0.0)'
 
     // Use JUnit test framework
-    testImplementation 'junit:junit:4.12'
+    testImplementation(platform('org.junit:junit-bom:5.10.0'))
+    testImplementation('org.junit.jupiter:junit-jupiter')
```
Comment (kinyoklion) [inline build.gradle:131]: I wanted some things from JUnit 5, so I moved forward.

File: .circleci/config.yml [hunk 3]
```diff
@@ -7,11 +7,6 @@ workflows:
   test:
     jobs:
       - build-linux
-      - test-linux:
```
Comment (kinyoklion) [inline .circleci/config.yml:10]: Dropping 1.8 as it isn't supported by the OF Sdk.

File: build.gradle [hunk 4]
```diff
@@ -19,6 +19,8 @@ java {
 repositories {
     mavenLocal()
     mavenCentral()
+    // Before LaunchDarkly release artifacts get synced to Maven Central they are here along with snapshots:
+    maven { url "https://oss.sonatype.org/content/groups/public/" }
```
Comment (kinyoklion) [inline build.gradle:23]: To allow using pre-releases when we want to.

File: build.gradle [hunk 5]
```diff
@@ -41,6 +43,15 @@ checkstyle {
     checkstyleTest.enabled = false
 }
 
+task generateJava(type: Copy) {
```
Comment (kinyoklion) [inline build.gradle:46]: The template code generation is so we have access to the version to put into the wrapper.

File: src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java [hunk 6]
```diff
@@ -86,41 +89,122 @@ public Metadata getMetadata() {
...
+                    emitProviderError(ProviderEventDetails.builder().message("Provider shutdown").build());
+                }
+            }
+        });
+        if (state == ProviderState.READY) {
+            return;
+        }
+
+        boolean initialized = client.getDataSourceStatusProvider().waitFor(DataSourceStatusProvider.State.VALID,
+            ChronoUnit.FOREVER.getDuration());
```
Comment (kinyoklion) [inline src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java:183]: Fun fact, FOREVER, is just the most digits that can put in without an overflow.

Comment (kinyoklion) [reply src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java:183]: waitFor also should return if we hit the "OFF" state from a terminal error.

File: src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java (hunk 1 above)
Comment (louis-launchdarkly) [reply src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java:92]: IDEs are weird.

File: src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java [hunk 7]
```diff
@@ -43,39 +47,38 @@ public String getName() {
...
+    public Provider(String sdkKey, LDConfig config) {
+        this(new LDClient(sdkKey, LDConfig.Builder.fromConfig(config)
+            .wrapper(Components.wrapperInfo()
+                .wrapperName("open-feature-java-server")
+                .wrapperVersion(Version.SDK_VERSION)).build()));
+    }
+
+    Provider(LDClientInterface client) {
+        this.client = client;
+        logger = client.getLogger();
```
Comment (kinyoklion) [inline src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java:78]: We added this in the migrations changes, which is really handy for OpenFeature.

File: src/main/java/com/launchdarkly/openfeature/serverprovider/ProviderConfiguration.java
Comment (kinyoklion) [inline src/main/java/com/launchdarkly/openfeature/serverprovider/ProviderConfiguration.java:1]: With the logger change we no longer need provider specific configuration, so this was deleted.

**Review by louis-launchdarkly — APPROVED** [review]

**Pull request merged**